analyses/cell-type-wilms-tumor-06/notebook/07_annotation_Across_Samples_exploration.Rmd

```diff
@@ -0,0 +1,586 @@
+
+return(p)
+}
+```
+
+
+#### do_Feature_boxplot
+
+`do_Feature_boxplot` shows boxplot of expression of a feature grouped by a metadata. 
+
+- `df` is the name of the table containing metadata and feature expression (counts) per cells
+- `group.by` is the name of the metadata to group the cells
+- `feature` is the name of the gene to average the expression
+
+```{r fig.width=10, fig.height=4, out.width='100%'}
+do_Feature_boxplot <- function(df, group.by, feature, split.by ){
+  
+  df <- df %>%
+    mutate(!!sym(group.by), group  = factor(!!sym(group.by), levels = c("fetal_nephron", "stroma", "immune", "endothelium")))
+    
+  p <- ggplot(df, 
+    aes(x = group, y = !!sym(feature), fill = group)
+  ) + 
+  geom_boxplot(size = 0.5, size.outlier = 0.25) +
+  facet_wrap(vars(!!sym(split.by)), scale = "free_y", ncol = 4) + 
+  theme_bw() +
+  theme(
+    axis.text.x = element_text(size = 12, angle = 0, hjust = 0.5, vjust = 0), 
+    legend.position = "none"
+  ) 
+
+return(p)
+}
+```
+
+## Analysis
+
+### Global cnv score
+
+As done in `06_cnv_infercnv_exploration.Rmd`, we calculate  single CNV score and assess its potential in identifying cells with CNV versus normal cells without CNV.
+
+We simply checked for each chromosome if the cell `has_cnv_chr`. 
+Would the cell have more than `cnv_threshold` chromosome with CNV, the global `has_cnv_score` will be TRUE.
+Else, the cell will have a `has_cnv_score` set to FALSE.
+
+
+
+
+```{r fig.width=10, fig.height=10, out.width='100%', results='asis'}
+
+#### CNV score
+  cell_type_df$has_cnv_score <- rowSums(cell_type_df[,grepl("has_cnv_chr", colnames(cell_type_df))])
+  cell_type_df$has_cnv_score[cell_type_df$has_cnv_score > params$cnv_threshold] <- "CNV"
+
+  # we know that some immune cells might have false positive CNV 
+  cell_type_df$has_cnv_score[cell_type_df$has_cnv_score %in% c(0:params$cnv_threshold)] <- "no CNV"
+
+
+table(cell_type_df$has_cnv_score)
+```   
+
+### First level annotation
+
+At first, we like to indicate in the `first.level_annotation` if a cell is normal, cancer or unknown.
+
+- _normal_ cells can be observe in all four compartments (`endothelium`, `immune`, `stroma` or `fetal nephron`) and do not have cnv. 
+We only allow a bit of flexibility in terms of cnv profile for immune and endothelium cells that have a high predicted score.
+Indeed, we know that false positive cnv can be observed in a cell type specific manner.
+
+The threshold used for the `predicted.score` is defined as a parameter of this notebook as `r params$predicted.celltype.threshold`.
+The threshold used for the identification of cnv is also defined in the params of the notebook as `r params$cnv_threshold`.
+
+- _cancer_ cells are either from the `stroma` or `fetal nephron` compartments and must have at least few cnv.
+
+
+
+```{r fig.width=10, fig.height=10, out.width='100%', results='asis'}
+
+cell_type_df$first.level_annotation <- "unknown"
+
+
+# Define normal cells
```

Comment:
One question I have here (and more generally for the notebook) is whether you want to use the scores when assigning labels based on results from label transfer.

For example, in your first condition here, you check whether a cell is fetal nephron or stroma. Cells which have scores of say 0.1 (aka, not very confident!) but are labeled nephron will be regarded as nephron, but there is an approach where you might say "any cell with a score below <some threshold> has an UNKNOWN compartment", and then not label these cells at all. This would be a separate condition: If the score for the compartment is less than a certain value, then keep that cell as unknown since we don't have reliable label transfer results.

That said, I don't think this matters quite so much for `compartment`, since those results are more reliable, but it may matter for the `cell_type` annotations from label transfer which has many more categories.

Reply:
This is a good idea, thanks!
I would however only apply it to "normal" cells, as we expect cancer cells to have lower `predicted.score`?

What about having at the end a quick check of the density of the `predicted.scores` for each of the `first/second.level_annotation` and filter out some of the annotations with too low confidency?

Reply:
>  I would however only apply it to "normal" cells, as we expect cancer cells to have lower predicted.score?

This makes sense to me, but please add a sentence that explicitly says this in the notebook about this expectation. I see you've added something about using the scores for normal cells (great!) so let's add this explanation too for why you don't use them for cells we're calling as cancer.

Reply:
> What about having at the end a quick check of the density of the predicted.scores for each of the first/second.level_annotation and filter out some of the annotations with too low confidency?

I think it would certainly be worth looking at the distribution of scores here, and then we can think about filtering. But, I might open a separate issue for this as something to circle back to after the deadline!